### PR TITLE
Add missing Report peer and Report profile photo commands

### PR DESCRIPTION
**Pull request by milichev**
This PR adds missing menu commands:

* Report peer (user/channel/group),
* Report profile picture.

The existing modal dialog is used for report values (reason and description).

The next missing API methods were generated:

*  `account.reportPeer`
*  `account.reportProfilePhoto`

### Screenshots

Report peer in the top top menu in the chat view:
Report peer in the top top menu in the chat view

Report Photo in the Profile picture viewer:
Report Photo in Profile picture viewer

Report peer in the left chat list:
Report peer in the left chat list

### Open questions

1. The logic of how the Report button is displayed in the profile avatar viewer mimics those rules in the mobile app: it is available when the avatar owner is available, and the current picture has an index > 0 (starting from the second). Does it make sense?

### CODE REVIEW

File: src/components/left/main/Chat.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ import ChatFolderModal from '../ChatFolderModal.async';
 import ChatCallStatus from './ChatCallStatus';
 
 import './Chat.scss';
+import ReportModal from '../../common/ReportModal';
```
Comment (Ajaxy) [inline src/components/left/main/Chat.tsx:55]: Please, fix imports order here.

File: src/components/middle/HeaderMenuContainer.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ import MenuItem from '../ui/MenuItem';
 import DeleteChatModal from '../common/DeleteChatModal';
 
 import './HeaderMenuContainer.scss';
+import ReportModal from '../common/ReportModal';
```
Comment (Ajaxy) [inline src/components/middle/HeaderMenuContainer.tsx:26]: Imports order.

File: src/global/types.ts [hunk 3]
```diff
@@ -615,6 +615,8 @@ export type NonTypedActionNames = (
   'setNewChatMembersDialogState' | 'disableHistoryAnimations' | 'setLeftColumnWidth' | 'resetLeftColumnWidth' |
   'openSeenByModal' | 'closeSeenByModal' | 'closeReactorListModal' | 'openReactorListModal' |
   'toggleStatistics' |
+  // accounts
+  'reportPeer' | 'reportProfilePhoto' |
```
Comment (Ajaxy) [inline src/global/types.ts:619]: Please, use typed `ActionPayloads` above for new actions.

File: src/global/actions/api/accounts.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { addActionHandler } from '../../index';
+import { selectChat } from '../../selectors';
+import { callApi } from '../../../api/gramjs';
+
+addActionHandler('reportPeer', async (global, actions, payload) => {
+  const {
+    chatId,
+    reason,
+    description,
+  } = payload!;
```
Comment (Ajaxy) [inline src/global/actions/api/accounts.ts:10]: No need for `!` once typed `ActionPayloads` is used.

File: src/global/actions/api/accounts.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  const result = await callApi('reportPeer', {
+    peer: chat,
+    reason,
+    description,
+  });
+
+  actions.showNotification({
+    message: result
+      ? 'Thank you! Your report will be reviewed by our team.'
```
Comment (Ajaxy) [inline src/global/actions/api/accounts.ts:25]: Is there a chance to use translated text? (`langProvider.ts:getTranslation`)

File: src/global/actions/api/messages.ts [hunk 6]
```diff
@@ -461,7 +461,7 @@ addActionHandler('reportMessages', async (global, actions, payload) => {
   actions.showNotification({
     message: result
       ? 'Thank you! Your report will be reviewed by our team.'
-      : 'Error occured while submiting report. Please, try again later.',
+      : 'Error occurred while submitting report. Please, try again later.',
```
Comment (Ajaxy) [inline src/global/actions/api/messages.ts:464]: 👍

File: src/api/gramjs/gramjsBuilders/index.ts [hunk 7]
```diff
@@ -340,6 +341,20 @@ export function buildChatPhotoForLocalDb(photo: GramJs.TypePhoto) {
   });
 }
 
+export function buildTypeInputPhoto(photo: ApiPhoto) {
```
Comment (Ajaxy) [inline src/api/gramjs/gramjsBuilders/index.ts:344]: → `buildInputPhoto`

**Review by Ajaxy — CHANGES_REQUESTED** [review]
Thanks for the PR, looks really good!

> The logic of how the Report button is displayed in the profile avatar viewer mimics those rules in the mobile app: it is available when the avatar owner is available, and the current picture has an index > 0 (starting from the second). Does it make sense?

Sounds about right.

**Comment by milichev** [comment]
@Ajaxy, thanks for the review!

Re `langProvider.ts:getTranslation`, is lang pack editing feasible? What's the workflow for adding new entries?

File: dev/localize/en-ios.json [hunk 8]
```diff
@@ -14254,6 +14254,10 @@
     "key": "ReportPeer.AlertSuccess",
     "value": "Thank you!\nYour report will be reviewed by our team."
   },
+  "ReportPeer.AlertFailure": {
```
Comment (Ajaxy) [inline dev/localize/en-ios.json:14257]: No need to edit these files, they are redundant and will be removed from the repo soon.

Comment (milichev) [reply dev/localize/en-ios.json:14257]: @Ajaxy, ok, there are non-localized fail cases as before.

**Review by Ajaxy — APPROVED** [review]

**Comment by Ajaxy** [comment]
Thanks for the PR, merged 🎉

**Pull request merged**